### ISSUE TITLE
Script Canvas: Networking types not compatible with scripting

### ISSUE THREAD

**Issue by lsemp3d** [o3de/o3de#7495]
Multiplayer::ClientInputId
Multiplayer::HostFrameId
Multiplayer::NetComponentId
Multiplayer::NetEntityId
Multiplayer::PropertyIndex
Multiplayer::RpcIndex

See #7351

**Comment by lmbr-pip** [comment]
For context from linked issue. "Types would need to change from uint16_t to AZ::u16 as those are type info specialized"

**Comment by AMZN-Olex** [comment]
@AMZN-Gene - could you take a look at this one?

**Comment by AMZN-Gene** [comment]
Absolutely! I dont think this will be blocking as those types aren't commonly used when scripting; except maybe net-entity-id

**Comment by gralco** [comment]
A workaround we have been using is to convert NetEntityIds (which has an underlying type of uint64_t) into strings for use in scripting. Since NetEntityIds are uint64_t and Lua/ScriptCanvas can't handle unsigned 64-bit ints, this made them unrelaible in scripting based on our testing (as expected); but converting to and from strings has worked well for us so far.

We convert them back-and-forth like so...

Getting the Multiplayer::NetEntityId and converting it into a string:
```
const Multiplayer::INetworkEntityManager* networkEntityManager = Multiplayer::GetMultiplayer()->GetNetworkEntityManager();
const Multiplayer::NetEntityId netEntityId = networkEntityManager->GetNetEntityIdById(entityId);
const AZStd::string netEntityIdStr = AZStd::to_string(netEntityId);
```
Converting a string back into Multiplayer::NetEntityId and getting the AZ::EntityId from it:
```
Multiplayer::NetEntityId netEntityId = static_cast<Multiplayer::NetEntityId>(AZStd::stoull(strNetEntityId));
const Multiplayer::INetworkEntityManager* networkEntityManager = Multiplayer::GetMultiplayer()->GetNetworkEntityManager();
const Multiplayer::ConstNetworkEntityHandle entity = networkEntityManager->GetEntity(netEntityId);
AZ::EntityId entityId = AZ::EntityId(AZ::EntityId::InvalidEntityId);
if (entity)
    entityId = entity.GetEntity()->GetId();
```

So until Multiplayer::NetEntityId is reflected/serialized this seems to be a viable workaround.
